Sprint №8, HomeWork

Vozzhd opened the pull request.

Vozzhd committed:
activity_main.xml is done
Added:
-dimens (defaults)
-icons(gearIcon,magnifier,noteIcon)
-colors (YA_blue)
Changed:
- strings.xml

Vozzhd committed:
activity_settings.xml is done

Added:
-icons
-activity_settings.xml
Changed:
-strings.xml
-dimens.xml
-colors.xml

Vozzhd committed:
HomeWork №2 is done

Added:
-buttons ID's
Changed:
MainActivity.kt

Vozzhd committed:
HomeWork №3 is done

Added:
-activity_search.xml and .kt
-activity_library.xml and .kt
-intents
Changed:
MainActivity.kt

Deleted:
-Toasts

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,19 @@
+
+        val backToMainButton = findViewById<ImageButton>(R.id.backButton)
+        backToMainButton.setOnClickListener{
+            startActivity(Intent(this, MainActivity::class.java))
```
Для перехода назад надо использовать не интент, а методы finish() или onBackPressed(), потому что иначе вместо возврата в предыдущий экран создается новый экземпляр MainActivity и при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на этот экран. На эту тему можно почитать про стек активити. Это будет критично в следующем спринте.

Vozzhd replied:
Заменил, спасибо за совет)

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,96 @@
+    android:orientation="vertical"
 
+    android:paddingLeft="16dp"
+    android:paddingRight="16dp"
```
paddingLeft + paddingRight = paddingHorizontal

Vozzhd replied:
Заменил, в работе

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,96 @@
-</androidx.constraintlayout.widget.ConstraintLayout>
\ No newline at end of file
+        android:layout_height="0dp"
+        android:layout_weight="1"
```
Лучше без веса, а просто wrap_content

Vozzhd replied:
Если без веса то эран начинает выглядеть не как в макете. 
В самом макете до первой кнопки расстояние 80\800, т.е 1\10. Потому и разнёс три кнопки с весом 3 каждая и 1 часть это вот этот тайтл.
![Вес от макета](https://github.com/Vozzhd/PlaylistMaker/assets/144613563/8b075ab0-3ed9-4f82-a0ee-8dc32f4bce9d)

faritowich replied:
нужно просто использовать отступы - padding или margin.

Vozzhd replied:
В смысле вообще от веса везде постараться уйти? 
Или речь ещё идёт именно об этом TextView?

faritowich replied:
речь именно об этом TextView

Vozzhd replied:
Ок, сделано, спасибо)

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,162 @@
+        <TextView
+            android:layout_width="0dp"
+            android:layout_height="match_parent"
+            android:layout_weight="1" />
```
Лишнее. Можно использовать FrameLayout и прижать текст и иконку к разным краям, без таких костылей

Vozzhd replied:
Заменил, убрал костыль, поставил всё на места.

faritowich commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,11 @@
+    <dimen name="defaultPadding">8dp</dimen>
+    <dimen name="defaultIconSize">24dp</dimen>
+    <dimen name="defaultMargin">16dp</dimen>
+    <dimen name="settingsLineHeight">61dp</dimen>
```
Рекомендуется делать размеры View и отступы кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

Vozzhd replied:
Размер 61 - исключительно из-за того что такой в макете.
![Высота из макета](https://github.com/Vozzhd/PlaylistMaker/assets/144613563/cc27ca8e-bfbc-407b-935d-c1b305b28db0)

faritowich replied:
Дизайнеры не всегда следуют рекомендациям гугла. А у нас задача сделать такой UI, чтобы он соответствовал макетам внешне, не обязательно пиксель в пиксель, тем более что это невозможно, учитывая огромное количество форм и размеров устройств. Так что если мы сделаем и UI как на макетах, и при этом последуем рекомендациям гугла, то будет круто

Vozzhd replied:
Ок, просто мы на учёбе и исходя из этого брал размеры)) поставлю 60
В теории - да, вроде отражён момент насчёт кратности, но вроде как он после этой вёрстки.

Vozzhd committed:
HomeWork №3 accepted
Changed after review:
-activity_settings.xml (LinearLayout -> FrameLayout)
-backButton in activity_settings.xml (Intent -> finish())
-padding (left-right -> horizontal)

Vozzhd committed:
HomeWork №3 accepted
Changed after review :
-activity_main.xml (ImageView weight -> height\padding)
-dimens.xml (61-60 for multiplicity = 4)

Vozzhd committed:
ДЗ №1 в прогрессе, промежуточный итог:
-работает переключение темы
Нужно доделать:
-исправить стили switchCompat
-замену иконок
-замену текста
